packages/commons/src/logging/loggerMiddleware.ts

```diff
@@ -22,44 +26,83 @@ const getLoggerMetadata = (): SessionMetaData => {
   const appContext = getContext();
   return !appContext
     ? {
-        userId: "",
-        organizationId: "",
-        correlationId: "",
+        userId: undefined,
+        organizationId: undefined,
+        correlationId: v4(),
+        eventType: undefined,
+        eventVersion: undefined,
+        streamId: undefined,
       }
     : {
-        userId: appContext.authData.userId,
-        organizationId: appContext.authData.organizationId,
+        userId: appContext.authData?.userId,
+        organizationId: appContext.authData?.organizationId,
         correlationId: appContext.correlationId,
+        eventType: appContext.messageData?.eventType,
+        eventVersion: appContext.messageData?.eventVersion,
+        streamId: appContext.messageData?.streamId,
       };
 };
 
 const logFormat = (
   msg: string,
   timestamp: string,
   level: string,
-  userId: string | undefined,
-  organizationId: string | undefined,
-  correlationId: string | undefined,
-  serviceName: string = ""
-) =>
-  `${timestamp} ${level.toUpperCase()} [${serviceName}] - [UID=${userId}] [OID=${organizationId}] [CID=${correlationId}] ${msg}`;
+  {
+    userId,
+    organizationId,
+    correlationId,
+    serviceName,
+    eventType,
+    eventVersion,
+    streamId,
+  }: {
+    userId: string | undefined;
+    organizationId: string | undefined;
+    correlationId: string | undefined;
+    serviceName: string | undefined;
+    eventType: string | undefined;
+    eventVersion: number | undefined;
+    streamId: string | undefined;
+  }
```

Comment:
what if we just use the context? 
```suggestion
  ctx: AppContext,
  serviceName: string | undefined
```

Reply:
I prefer a flat data structure in this case. Expecially if we decide to separate the `AppContext` into two different type (one for the Producer and one for the Consumer)

Reply:
may I ask you why?

Reply:
If we split the `AppContext` because here we need data from different contexts (the ApiContext and the ConsumerContext) and I think it's more readable to have a flat input parameter structure, but it's more a personal preference.

Reply:
I have the opposite feeling about this 😄 because we:
- use a specific type, which is not only for convenience, but it's actually what the function needs (the context)
- define when the function should expect each field (with the union type)
- replace 16 rows of parameters definition with just 2 🤣 

But we can move this refactor along with the addition of specific context